From f53704f164f67c1b9e3e196e642afd24319f64d6 Mon Sep 17 00:00:00 2001
From: Haim Boot <hayim@marvell.com>
Date: Thu, 5 May 2016 14:20:04 +0300
Subject: [PATCH 1091/1240] dts: a70x0: add all pci device tree configuration

add dts for all pci configuration (with id of 4)

Change-Id: I63d04efe3b6492534d09359fcdce7aaa789b0dda
Signed-off-by: Haim Boot <hayim@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29553
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Reviewed-by: Yuval Caduri <cyuval@marvell.com>
---
 arch/arm/dts/Makefile                    |   5 +-
 arch/arm/dts/armada-70x0-db_all_pcie.dts | 190 +++++++++++++++++++++++++++++++
 2 files changed, 193 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm/dts/armada-70x0-db_all_pcie.dts

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index ad711e7..5e569bf 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -83,8 +83,9 @@ dtb-$(CONFIG_DEVEL_BOARD) += armada-7040-rz-db_nand.dtb \
 			     armada-7040-rz-db_rxaui.dtb
 else
 dtb-y += armada-70x0-customer0.dtb
-dtb-$(CONFIG_DEVEL_BOARD) += armada-70x0-db_router.dtb \
-			     armada-70x0-db_kr.dtb
+dtb-$(CONFIG_DEVEL_BOARD) += armada-70x0-db_router.dtb 	\
+			     armada-70x0-db_kr.dtb	\
+			     armada-70x0-db_all_pcie.dtb
 endif
 endif #CONFIG_TARGET_ARMADA_8K
 
diff --git a/arch/arm/dts/armada-70x0-db_all_pcie.dts b/arch/arm/dts/armada-70x0-db_all_pcie.dts
new file mode 100644
index 0000000..da8b30d
--- /dev/null
+++ b/arch/arm/dts/armada-70x0-db_all_pcie.dts
@@ -0,0 +1,190 @@
+/*
+ * ***************************************************************************
+ * Copyright (C) 2015 Marvell International Ltd.
+ * ***************************************************************************
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation, either version 2 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ * ***************************************************************************
+ */
+
+/dts-v1/;
+
+#include "armada-70x0.dtsi" /* include SoC device tree */
+#include "armada-70x0-dev-info.dtsi" /* include Marvell specific info (S@R, MPP cmd) */
+
+/ {
+	model = "DB-ARMADA-7040-A0 (ALL_PCI)";
+	compatible = "marvell,armada-70x0-db", "marvell,armada-70x0";
+	fdt_config_id = <4>;
+	board_id = <0x12>;
+
+	ap-806 {
+		internal-regs {
+			spi0: spi@510600 {
+				status = "disabled";
+			};
+			i2c0: i2c@511000 {
+				status = "disabled";
+			};
+			pinctl: pinctl@6F4000 {
+					/* MPP Bus:
+						SDIO  [0-10]
+						UART0 [11,19]
+					*/
+					  /* 0 1 2 3 4 5 6 7 8 9 */
+				pin-func = < 1 1 1 1 1 1 1 1 1 1
+					     1 3 0 0 0 0 0 0 0 3>;
+			};
+			mmc0: mmc@6E0000 {
+				status = "okay";
+			};
+		};
+	};
+	cp110-master {
+		internal-regs {
+			spi0@700600 {
+				status = "disabled";
+			};
+			spi1@700680 {
+				status = "okay";
+			};
+			i2c@701000 {
+				status = "okay";
+			};
+			i2c@211000 { /* MSS_i2c0 */
+				status = "disabled";
+			};
+			i2c@211100 { /* MSS_i2c1 */
+				status = "disabled";
+			};
+			usb3@500000 {
+				status = "disabled";
+			};
+			usb3@510000 {
+				status = "disabled";
+			};
+			sata@540000 {
+				status = "disabled";
+			};
+			mmc@780000 {
+				status = "disabled";
+			};
+			pcie-controller {
+				pcie@0,0 {
+					status = "okay";
+				};
+				pcie@1,0 {
+					status = "okay";
+				};
+				pcie@2,0 {
+					status = "okay";
+				};
+			};
+			nand@720000 {
+				status = "disabled";
+			};
+
+			pinctl@440000 {
+					/* MPP Bus:
+					   RGMII0[0-11]
+					   SPI   [13-16]
+					   SATA1 [28]
+					   UART0 [29-30]
+					   SMI	 [32,34]
+					   SATA0 [36]
+					   I2C	 [37-38]
+					   UART1 [40-41]
+					   XSMI	 [42-43]
+					   RGMII1[44-55]
+					   SD	 [56-62]
+					*/
+					/*   0   1   2   3   4   5   6   7   8   9 */
+				pin-func = < 3   3   3   3   3   3   3   3   3   3
+					     3   3   0   3   3   3   3   0   0   0
+					     0   0   0   0   0   0   0   0   9   0xA
+					     0xA 0   7   0   7   0   6   2   2   0
+					     7   7   8   8   1   1   1   1   1   1
+					     1 	 1   1   1   1   1   0xE 0xE 0xE 0xE
+					     0xE 0xE 0xE>;
+			};
+
+			comphy {
+				/* Serdes Configuration:
+					Lane 0-3: PEX0x4
+					Lane 4:   PEX1x1
+					Lane 5:   PEX2x1
+				*/
+				phy0 {
+					phy-type = <PHY_TYPE_PEX0>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy1 {
+					phy-type = <PHY_TYPE_PEX0>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy2 {
+					phy-type = <PHY_TYPE_PEX0>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy3 {
+					phy-type = <PHY_TYPE_PEX0>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy4 {
+					phy-type = <PHY_TYPE_PEX1>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy5 {
+					phy-type = <PHY_TYPE_PEX2>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+			};
+			mdio {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				mac0 {
+					phy-mode = "sgmii";
+					status = "disabled";
+				};
+				mac2 {
+					phy-mode = "rgmii"; /* rgmii-0 */
+					phy = <&phy2>;
+					status = "okay";
+				};
+				mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy3>;
+					status = "okay";
+				};
+			};
+			ppv22@000000 {
+				eth0@010000 {
+					status = "disabled";
+				};
+				eth1@020000 {
+					status = "okay";
+				};
+				eth2@030000 {
+					status = "okay";
+				};
+			};
+		};
+	};
+};
-- 
1.9.1

